pkg/operator/controller/controller_dns_configmap_test.go

```diff
@@ -883,6 +914,84 @@ func TestDesiredDNSConfigmapUpstreamResolvers(t *testing.T) {
 	}
 }
 
+func Test_coreDNSCache(t *testing.T) {
+	testcases := []struct {
```

Comment:
Please add test cases for fractional values.  For example:
```go
		{
			name: "999ms rounds to 0, resulting in the default settings",
			dns: &operatorv1.DNS{
				Spec: operatorv1.DNSSpec{
					Cache: operatorv1.DNSCache{
						PositiveTTL: metav1.Duration{999 * time.Millisecond},
						NegativeTTL: metav1.Duration{999 * time.Millisecond},
					},
				},
			},
			expectedPTTL: cacheDefaultMaxPositiveTTLSeconds,
			expectedNTTL: cacheDefaultMaxNegativeTTLSeconds,
		},
		{
			name: "1999ms rounds to 1s",
			dns: &operatorv1.DNS{
				Spec: operatorv1.DNSSpec{
					Cache: operatorv1.DNSCache{
						PositiveTTL: metav1.Duration{1999 * time.Millisecond},
						NegativeTTL: metav1.Duration{1999 * time.Millisecond},
					},
				},
			},
			expectedPTTL: 1,
			expectedNTTL: 1,
		},
```

Reply:
The minimum unit allowed by the API is seconds so are these tests a preventative measure? I tested some values [here](https://github.com/openshift/api/pull/1214#issuecomment-1215700550) per the API team's request.

Reply:
Right, I'm looking for something along those lines, the goal being to codify the expected behavior around fractional values.  For example, a reasonable person might expect "0.9s" to be clamped or rounded up to 1s, but instead "0.9s" gets rounded down to 0 and thus defaulted to 30s, so the unit tests should codify that behavior.  

I see you also tested various ways of expressing zero values.  For the purposes of this unit test, we don't need to worry about the various yaml representations of zero values since the corresponding Go values are all equal—that is, the following test will pass (this is just to illustrate my point; I'm not asking you to add the following to your PR):
```go
	a := operatorv1.DNS{
		Spec: operatorv1.DNSSpec{
			Cache: operatorv1.DNSCache{
				PositiveTTL: metav1.Duration{},
				NegativeTTL: metav1.Duration{},
			},
		},
	}
	b := operatorv1.DNS{
		Spec: operatorv1.DNSSpec{
			Cache: operatorv1.DNSCache{
				PositiveTTL: metav1.Duration{0 * time.Second},
				NegativeTTL: metav1.Duration{0 * time.Second},
			},
		},
	}
	c := operatorv1.DNS{
		Spec: operatorv1.DNSSpec{
			Cache: operatorv1.DNSCache{},
		},
	}
	if !reflect.DeepEqual(a, b) {
		t.Errorf("a!=b\na: %v\nb: %v", a, b)
	}
	if !reflect.DeepEqual(a, c) {
		t.Errorf("a!=c\na: %v\nc: %v", a, c)
	}
```

Reply:
Added a few more test cases in the latest changes.